File: Leetcode_Java/isPalindrome.java
package Leetcode_Java;

class Solution {

    private boolean isAlphaNum(char c) {
        return (('A' <= c && c <= 'Z') || ('a' <= c && c <= 'z') || ('0' <= c && c <= '9'));
    }

    public boolean isPalindrome(String s) {
        int left = 0;
        int right = s.length() - 1;

        while (left < right) {
            while (left < right && !isAlphaNum(s.charAt(left))) {
                left++;
            }

            while (left < right && !isAlphaNum(s.charAt(right))) {
                right--;
            }

            if (left < right && Character.toLowerCase(s.charAt(left)) != Character.toLowerCase(s.charAt(right))) {
                return false;
            }

            left++;
            right--;
        }

        return true;
        
    }
}

File: Leetcode_Python/isPalindrome.py
class Solution:
    def isAlphaNum(self, c):
        return ((ord("A") <= ord(c) <= ord("Z")) or (ord("a") <= ord(c) <= ord("z")) or (ord("0") <= ord(c) <= ord("9")))

    def isPalindrome(self, s: str) -> bool:
        left = 0
        right = len(s) - 1

        
        while left < right:
            while left < right and not self.isAlphaNum(s[left]):
                left += 1
        
            while left < right and not self.isAlphaNum(s[right]):
                right -= 1
            
            while left < right and s[left].lower() != s[right].lower():
                return False
            
            left += 1
            right -= 1

            
        return True
